src/utilities/ci.jl

```diff
@@ -16,9 +17,29 @@ clone_hostname(::GitHubActions) = "github.com"
 github_repository(env::AbstractDict=ENV) = env["GITHUB_REPOSITORY"]
 github_token(env::AbstractDict=ENV) = env["GITHUB_TOKEN"]
 
+struct GitLabCI <: CIService
+    username::String
+    email::String
+end
+
+function GitLabCI()
+    # TODO: Not sure what to actually put here
+    return GitLabCI(
+        "gitlab-ci[bot]", "gitlab-ci[bot]@gitlab.com"
+    )
+end
+
+api_hostname(::GitLabCI) = "https://gitlab.com/api/v4"
+clone_hostname(::GitLabCI) = "gitlab.com"
+
+gitlab_repository(env::AbstractDict=ENV) = env["CI_PROJECT_PATH"]
+gitlab_token(env::AbstractDict=ENV) = env["GITLAB_TOKEN"]
```

Comment:
```suggestion
repository(::GitLabCI, env::AbstractDict=ENV) = env["CI_PROJECT_PATH"]
token(:GitLabCI, env::AbstractDict=ENV) = env["GITLAB_TOKEN"]
```
and similar changed to the `GitHubCI` bit?

Reply:
ah yeah that makes sense, with the other change to that if block above, I think this would be nicer.